wasm/jit/jit_arm64.go

```diff
@@ -476,20 +512,141 @@ func (c *arm64Compiler) compileLabel(o *wazeroir.OperationLabel) (skipThisLabel
 	return false
 }
 
+// compileUnreachable implements compiler.compileUnreachable for the arm64 architecture.
 func (c *arm64Compiler) compileUnreachable() error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.exit(jitCallStatusCodeUnreachable)
 }
 
+// compileSwap implements compiler.compileSwap for the arm64 architecture.
 func (c *arm64Compiler) compileSwap(o *wazeroir.OperationSwap) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	x := c.locationStack.peek()
+	y := c.locationStack.stack[int(c.locationStack.sp)-1-o.Depth]
+
+	if err := c.maybeCompileEnsureOnGeneralPurposeRegister(x); err != nil {
```

Comment:
are maybe and ensure redundant? I think ensure was used to say "unless already" which is same as maybe unless I am missing something